lib/widget/button.dart

```diff
@@ -82,20 +90,39 @@ class ButtonState extends WidgetState<Button> {
   @override
   Widget buildWidget(BuildContext context) {
     bool isOutlineButton = widget._controller.outline ?? false;
-    
-    Text label = Text(Utils.translate(widget._controller.label ?? '', context));
+
+    ScopeManager? scopeManager = DataScopeWidget.getScope(context);
+    Widget? startingIcon;
+    Widget? endingIcon;
+    if (widget._controller.startingIcon != null) {
+      startingIcon = scopeManager?.buildWidgetFromDefinition(
+          widget._controller.startingIcon);
+    }
+    if (widget._controller.endingIcon != null) {
+      endingIcon = scopeManager?.buildWidgetFromDefinition(widget._controller.endingIcon);
+    }
+    List<Widget> labelParts = [Text(Utils.translate(widget._controller.label ?? '', context))];
+    if (startingIcon != null) {
+      labelParts.insertAll(0, [startingIcon]);
+    }
+    if (endingIcon != null) {
+      labelParts.addAll([endingIcon]);
+    }
+    Widget labelLayout = Row(
```

Comment:
@vusters it seems like Row is required to render icons after the button and have the ability to have both icons. Another benefit is Row automatically responds to the device text direction, i.e. RTL. Let me know if this is concerning to you or you think we should move in a different direction.

Reply:
This is fine.